.github/workflows/docker.yml

```diff
@@ -0,0 +1,78 @@
+name: Telescope Docker
+
+# Cancel workflow if there is a new change to the branch.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+on:
+  push:
+    branches: [main]
+    tags:
+      - "*"
+  merge_group:
+  pull_request:
+    branches:
+      - main
+      - 'feature/**'
+      - 'v[0-9]+.[0-9]+'
+    # This needs to be declared explicitly so that the job is actually
+    # run when moved out of draft.
+    types: [opened, synchronize, reopened, ready_for_review]
+
+
+jobs:
+  build-and-push-docker-image:
+    # This is done to limit the runner cost.
+    # if: github.event.pull_request.draft == false
+    name: Build images and push to docker hub
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - name: List cached docker images
+        run: docker image ls
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+        
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+        with:
+            platforms: all
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Docker Meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: blockopsnetwork/telescope
```

Comment:
Set this as Variable in a MAkefile

Reply:
Sure, but i don't see how it can be reused here.

Reply:
unless i'm missing something, i don't think we need to reset:

https://github.com/blockopsnetwork/telescope/blob/497a62462873724f19452dc6338e6b54d2d0eafa/Makefile#L3

Reply:
So it falls back to this blockopsnetwork/telescope:v0.1.5 if AGENT_IMAGE env is not set, so that means in your workflow, you can just set AGENT_IMAGE directly or make the IMAGE static and TAG dynamic...do you get?

Reply:
yeah, i get the fallback logic but i think its a bit redundant, 
because we also kinda have a static tag https://github.com/blockopsnetwork/telescope/blob/43391ad618977a133f519e84450219fa78446a43/.github/workflows/docker.yml#L60

This basically means we'll always have an blockopsnetwork/telescope:yy-M-D-commit-hash i.e something like this `blockopsnetwork/telescope:2024-06-24-1d0217f` then the actual release tag or `edge` tags depending on what triggers the workflow, they're done here:  https://github.com/blockopsnetwork/telescope/blob/43391ad618977a133f519e84450219fa78446a43/.github/workflows/docker.yml#L61-L63

Reply:
@haroldsphinx checkout https://hub.docker.com/r/blockopsnetwork/telescope/tags